pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -60,10 +59,23 @@ func (v *ControlPlaneMutator) Handle(ctx context.Context, req atypes.Request) at
 	// on create we set the version to the current default version
 	// on update we leave the version intact to preserve the v1.0 version
 	// implied by the missing version field, which we added in version v1.1
-	if smcp.Spec.Version == "" && req.AdmissionRequest.Operation == admissionv1beta1.Create {
-		log.Info("Setting .spec.version to default value", "version", common.DefaultMaistraVersion)
-		newSmcp.Spec.Version = common.DefaultMaistraVersion
-		smcpMutated = true
+	if smcp.Spec.Version == "" {
+		switch req.AdmissionRequest.Operation {
+		case admissionv1beta1.Create:
+			log.Info("Setting .spec.version to default value", "version", maistra.DefaultVersion.String())
+			newSmcp.Spec.Version = maistra.DefaultVersion.String()
+			smcpMutated = true
+		case admissionv1beta1.Update:
+			if len(smcp.Status.AppliedVersion) == 0 {
+				// this must have been created before 1.1
+				newSmcp.Spec.Version = maistra.LegacyVersion.String()
+			} else {
+				// don't change the version
+				newSmcp.Spec.Version = smcp.Status.AppliedVersion
+			}
+			log.Info("Setting .spec.version to default value", "version", maistra.LegacyVersion.String())
+			smcpMutated = true
+		}
```

Comment:
What if the user removes the `Version` field during an update?  I think this will need to be reset to the previously `AppliedVersion` or else a `1.0` control plane will be installed, whether they want that or not.  Regarding the legacy initialization, I think it does help, as currently, reconciliation only occurs when the SMCP resource has been modified, so this would simplify reconciliation.  However, if we add code to trigger reconciliation in response to user modifications of managed resources, we would need to ensure empty values for `Version` are supported in the reconciler.

Reply:
Oh, I wouldn't allow unsetting the version, once it's set. I mean, the mutating webhook should default the version to the one in the oldObject (if set).

Also, I do think we'll start triggering reconciliation much more frequently than we do now. It's really not ok to fix the owned resources only when the user updates the SMCP. That goes against the basic idea of declarative configuration.

Reply:
Okay, I'll modify this to initialize it to the `oldSmcp.Spec.Version` and leave it at that.  This will mean that `Version == ""` will imply `v1.0`.

Reply:
We could add a `GetEffectiveVersion()` to the SMCP struct to centralize this, but I don't know how we'd force ourselves to use the function instead of simply reading the field.